db/migrate/20150403173410_add_has_joined_to_cud.rb

```diff
@@ -0,0 +1,27 @@
+class AddHasJoinedToCud < ActiveRecord::Migration
+
+  def up
+
+    add_column :courses, :public, :boolean
+    add_column :courses, :requires_permission, :boolean
```

Comment:
I'd imagine you would want to set public to false and requires_permission to true for both of these columns.  (Maybe make that the default as well?)

And unrelated, but something you should know, adding three columns like this essentially copies the table three times, each time with one added column.  Since `:courses` is very small, it doesn't matter, don't change it, but for something like `:scores`, you should create a new table with all three columns added, and then copy rows, one by one.